File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/mcp/docs/completions/prompts/MyPromptsCompletions.kt
package io.micronaut.mcp.docs.completions.prompts

//tag::imports[]
import io.micronaut.context.annotation.Requires
import io.micronaut.mcp.annotations.PromptCompletion
import jakarta.inject.Singleton

//end::imports[]
@Requires(property = "spec.name", value = "MyPromptsCompletionsTest")
//tag::clazz[]
@Singleton
class MyPromptsCompletions {
    @PromptCompletion(name = "code_review")
    fun languages(language: String?): List<String> {
        if (language != null && language.startsWith("py")) {
            return listOf("python", "pytorch", "pyside")
        }
        return emptyList()
    }
}
//end::clazz[]

File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/completions/prompts/MyPromptsCompletions.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.mcp.annotations import PromptCompletion
# end::imports[]


@Requires(property="spec.name", value="MyPromptsCompletionsTest")
# tag::clazz[]
@Singleton
class MyPromptsCompletions:
    @PromptCompletion(name="code_review")
    def languages(self, language: str | None) -> list[str]:
        if language is not None and language.startswith("py"):
            return ["python", "pytorch", "pyside"]
        return []
# end::clazz[]
